In [1]:
hyperparameters = {
    'epochs': [200, 400, 600],
    'lr': [0.001, 0.01, 0.1],
    'hidden': [128, 256, 512],
    'weight_decay': [1e-4, 5e-4, 1e-3],
    'dropout': [0.4, 0.6, 0.8],
    'alpha': [1.0, 2.0, 3.0],
    'beta': [1.0, 2.0, 3.0],
    'batch_size': [1024, 2048, 4096],
    'order': [1, 2, 3],
    'tau': [0.5, 1.0, 1.5],
    'k': [10, 15, 20]
}

In [3]:
for hyperparam, values in hyperparameters.items():
    for value in values:
        command = f'python train.py --no-cuda --{hyperparam} {value}'
        print(f'Running: {command}')
        !{command}

Running: python train.py --no-cuda --epochs 200
balance: 0.8034188034188035
entropy: 2.0600922182172106
nmi: 0.9545096825324375
acc: 0.927
Running: python train.py --no-cuda --epochs 400
balance: 0.8205128205128205
entropy: 2.0585630749954986
nmi: 0.9560432879440717
acc: 0.926
Running: python train.py --no-cuda --epochs 600
balance: 0.8103448275862069
entropy: 2.0573245473867807
nmi: 0.9557601145992054
acc: 0.926
Running: python train.py --no-cuda --lr 0.001
balance: 0.8166666666666667Running: python train.py --no-cuda --lr 0.01

entropy: 1.3747240689103184
nmi: 0.7757445742914731
acc: 0.696
balance: 0.13333333333333333
entropy: 2.165432209046507
nmi: 0.7904152465456173
acc: 0.745
Running: python train.py --no-cuda --lr 0.1
balance: 0
entropy: 2.0226148323278874
nmi: 0.5362254097505867
acc: 0.533
Running: python train.py --no-cuda --hidden 128
balance: 0.8103448275862069
entropy: 2.05864660542343
nmi: 0.9606492075674866
acc: 0.928
Running: python train.py --no-cuda --hidden 256
balance

Traceback (most recent call last):
  File "c:\Users\Idries\Desktop\FACT2024\FACT-2024\Fair-Clustering-Codebase\Consensus-Fair-Clustering\train.py", line 85, in <module>
    Y[i,l] = 1.0
    ~^^^^^
IndexError: index 6 is out of bounds for axis 1 with size 5


balance: 0.8173913043478261
entropy: 2.0610231329906927
nmi: 0.949313048666878
acc: 0.921
Running: python train.py --no-cuda --k 15
balance: 0.8709677419354839
entropy: 2.0642325262084666
nmi: 0.7880850228758991
acc: 0.791


model without dropout
balance: 0.8103448275862069
entropy: 2.405305485981072
nmi: 1.0
acc: 1.0

model without layer norm
balance: 0.8503937007874016
entropy: 2.061310947823611
nmi: 0.8458033286683967
acc: 0.825

model baseline
balance: 0.7136150234741784
entropy: 2.0217480310141487
nmi: 0.7572925091077589
acc: 0.71

with sigmoid act fn instead of GeLu
balance: 0.8130081300813008
entropy: 2.0610110061369324
nmi: 0.9525717154451236
acc: 0.928

hidden dimension 128:
balance: 0.8103448275862069
entropy: 2.064726162802119
nmi: 0.7933344870509539
acc: 0.763

In [2]:
for hyperparam, values in hyperparameters.items():
    for value in values:
        command = f'python train.py --data cora_balanced --no-cuda --seed 42 --{hyperparam} {value}'
        print(f'Running: {command}')
        !{command}

Running: python train.py --data cora_balanced --no-cuda --seed 42 --epochs 200
balance: 0.8103448275862069
entropy: 2.0632360401458376
nmi: 0.8222833410346752
acc: 0.771
Running: python train.py --data cora_balanced --no-cuda --seed 42 --epochs 400
balance: 0.8103448275862069
entropy: 2.0623500524715586
nmi: 0.8282685738852996
acc: 0.773
Running: python train.py --data cora_balanced --no-cuda --seed 42 --epochs 600
balance: 0.8103448275862069
entropy: 2.0619968984117616
nmi: 0.8299782152865893
acc: 0.774
Running: python train.py --data cora_balanced --no-cuda --seed 42 --lr 0.001
balance: 0.5517241379310345
entropy: 1.6693895532544134
nmi: 0.7724023177474226
acc: 0.711
Running: python train.py --data cora_balanced --no-cuda --seed 42 --lr 0.01
balance: 0.8103448275862069
entropy: 2.0623500524715586
nmi: 0.8282685738852996
acc: 0.773
Running: python train.py --data cora_balanced --no-cuda --seed 42 --lr 0.1
balance: 0.8658536585365854
entropy: 1.7204669504237833
nmi: 0.7718356893297511


Traceback (most recent call last):
  File "c:\Users\Idries\Desktop\FACT2024\FACT-2024\Fair-Clustering-Codebase\Consensus-Fair-Clustering\train.py", line 92, in <module>
    Y[i,l] = 1.0
    ~^^^^^
IndexError: index 6 is out of bounds for axis 1 with size 6


balance: 0.8103448275862069
entropy: 2.0623500524715586
nmi: 0.8282685738852996
acc: 0.773
Running: python train.py --data cora_balanced --no-cuda --seed 42 --k 15
balance: 0
entropy: 1.7202672898066997
nmi: 0.846658887572819
acc: 0.834


In [4]:
command = f'python train.py --data cora_balanced --no-cuda --seed 42 --k 20'
print(f'Running: {command}')
!{command}

Running: python train.py --data cora_balanced --no-cuda --seed 42 --k 20
balance: 0.8215767634854771
entropy: 1.7163603663232783
nmi: 0.7706933469445052
acc: 0.747


In [2]:
for hyperparam, values in hyperparameters.items():
    for value in values:
        command = f'python train.py --data cora_balanced --no-cuda --seed 43 --{hyperparam} {value}'
        print(f'Running: {command}')
        !{command}

Running: python train.py --data cora_balanced --no-cuda --seed 43 --epochs 200
balance: 0.7729468599033816
entropy: 1.6993039016957516
nmi: 0.8129169385775964
acc: 0.767
Running: python train.py --data cora_balanced --no-cuda --seed 43 --epochs 400
balance: 0
entropy: 1.9237657126504502
nmi: 0.7654416395144711
acc: 0.708
Running: python train.py --data cora_balanced --no-cuda --seed 43 --epochs 600
balance: 0.26506024096385544
entropy: 2.240130450607018
nmi: 0.7887817290524752
acc: 0.743
Running: python train.py --data cora_balanced --no-cuda --seed 43 --lr 0.001
balance: 0.8264462809917356
entropy: 1.374145575076511
nmi: 0.7759478666860784
acc: 0.693
Running: python train.py --data cora_balanced --no-cuda --seed 43 --lr 0.01
balance: 0
entropy: 1.9237657126504502
nmi: 0.7654416395144711
acc: 0.708
Running: python train.py --data cora_balanced --no-cuda --seed 43 --lr 0.1
balance: 0.6060606060606061
entropy: 3.0564335116387675
nmi: 0.9311091643080242
acc: 0.931
Running: python train.py

In [3]:
for hyperparam, values in hyperparameters.items():
    for value in values:
        command = f'python train.py --data cora_balanced --no-cuda --seed 44 --{hyperparam} {value}'
        print(f'Running: {command}')
        !{command}

Running: python train.py --data cora_balanced --no-cuda --seed 44 --epochs 200
balance: 0.8034188034188035
entropy: 2.061110218493689
nmi: 0.8252370831744583
acc: 0.772
Running: python train.py --data cora_balanced --no-cuda --seed 44 --epochs 400
balance: 0.8034188034188035Running: python train.py --data cora_balanced --no-cuda --seed 44 --epochs 600

entropy: 2.0607286947892334
nmi: 0.8291241909481524
acc: 0.776
balance: 0.8103448275862069
entropy: 2.0618799839033874
nmi: 0.8338974108758488
acc: 0.778
Running: python train.py --data cora_balanced --no-cuda --seed 44 --lr 0.001
balance: 0.8333333333333334
entropy: 1.3764122404221781
nmi: 0.7542171679791688
acc: 0.691
Running: python train.py --data cora_balanced --no-cuda --seed 44 --lr 0.01
balance: 0.8034188034188035Running: python train.py --data cora_balanced --no-cuda --seed 44 --lr 0.1

entropy: 2.0607286947892334
nmi: 0.8291241909481524
acc: 0.776
balance: 0
entropy: 2.7322592046845413
nmi: 0.9309617726516256
acc: 0.931
Running